docs/DeveloperGuide.adoc

```diff
@@ -686,8 +685,126 @@ account for cases such as the guest having a personalised discount due to
 the usage of vouchers or certain credit cards. Thus, ExpenseType does not
 need to be modified by users in the application. Nonetheless, it is still
 possible to modify the default information through editing the XML file.
+
+===== Aspect: Assignment of the Expenses object
+* **Alternative 1 (current choice):** Assign `expenses` to each `Room`.
+** Pros: Suitable for current architecture. Each `Booking` only has one `Guest`,
+   and each `Guest` will only stay in one `Room`. Makes more sense to assign to
+   `Room` such that it represents the expenditure of the entire `Room` and not
+   one `Guest`, since the occupants of the `Room` can only contribute to one single
+   `Expenses` object. `Room` is a more natural choice over `Booking` as `Booking`
+   is meant to encapsulate booking information such as timing and `Room`.
+   Not much difference in implementation no matter which one of the three classes
+   it is assigned to.
+** Cons: May be confusing to implement. Need to ensure that there are no expenses
+   for rooms that have no guests.
+* **Alternative 2:** Assign `expenses` to each `Guest`.
```

Comment:
You can also say that this requires tracking expenses across different rooms for a guest, if suppose he has multiple bookings at once.